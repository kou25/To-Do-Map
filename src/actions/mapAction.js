import {GET_MAP, LOADING} from './types'
import axios from 'axios'

export const getMap=()=>async dispatch=>{
    const res= await axios.get('https://docs.mapbox.com/mapbox-gl-js/assets/earthquakes.geojson');
    dispatch({
        type: GET_MAP,
        payload: res.data
    })
}

export const setLoading = () => {
    return {
      type: LOADING
    };
  };