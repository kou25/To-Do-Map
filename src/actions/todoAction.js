import { GET_TODOS, DELETE_TODO,
     ADD_TODO,GET_TODO, UPDATE_TODO, LOADING } from './types';
import axios from 'axios'

export const getTodos=()=>async dispatch=>{
    const res= await axios.get('http://localhost:3333/todos');
    dispatch({
        type: GET_TODOS,
        payload: res.data
    })
}

export const deleteTodo=(id)=>async dispatch=>{
    try{
    await axios.delete(`http://localhost:3333/todos/${id}`)
    dispatch({
        type: DELETE_TODO,
        payload: id 
    })
}catch (e) {
    dispatch({
      type: DELETE_TODO,
      payload: id
    });
  }
}

export const addTodo=(todo)=>async dispatch=>{
    const res=await axios.post('http://localhost:3333/todos', todo)
    dispatch({
        type: ADD_TODO,
        payload: res.data
    })
}

export const getTodo= id => async dispatch=>{
    const res = await axios.get(`http://localhost:3333/todos/${id}`)
    dispatch({
        type: GET_TODO,
        payload: res.data
    })
}

export const editTodo=todo=>async dispatch=>{
    const res = await axios.put(`http://localhost:3333/todos/${todo.id}`,todo)
    dispatch({
        type: UPDATE_TODO,
        payload: res.data
    })
}

export const setLoading = () => {
    return {
      type: LOADING
    };
  };