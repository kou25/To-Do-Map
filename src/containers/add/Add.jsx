import React, { Component } from 'react'
import {connect} from 'react-redux'
import PropTypes from 'prop-types';
import {addTodo} from '../../actions/todoAction'


class Add extends Component {
    state={
        name:'',
        desc:''
    }
    onSubmit=(e)=> {
        e.preventDefault();

        const {name, desc}=this.state
        
    const newTodo={
        name,
        desc
    }

    this.props.addTodo(newTodo)
    // Clear State
    this.setState({
      name: '',
      desc:''
    });

    this.props.history.push('/view');
    }
    onChange=e=>this.setState({[e.target.name]: e.target.value})
    render() {
        const{name,desc}=this.state
        return (
            <div className="container">
                <div className="d-sm-flex align-items-center justify-content-between mb-4">
                    <h1 data-testid="heading" style={{color:"#8B4513",textShadow:"1px 1px 2px black",  fontFamily:"cursive", margin:"50px 30%"}}>
                        Add To-DO<i className="fa fa-plus-circle" aria-hidden="true" style={{marginLeft:"20px", color:"#8B4513"}}></i></h1>
                </div>
                <div className="card shadow mb-2"  style={{width: "40%", fontFamily: "cursive", borderRadius: "20px", marginLeft:"22%"}}>
                    <div className="card-body">
                    <form  onSubmit={this.onSubmit}>
                            <div className="form-group">
                                <label htmlFor="name">Pads</label>
                                <input type="text" 
                                data-testid="name"
                                name="name"
                                className="form-control" 
                                placeholder="Enter name for to-do"
                                value={name}
                                required
                                onChange={this.onChange}/>
                               </div>
                               <div className="form-group">
                                    <label htmlFor="description">To Dos</label>
                                    <br/>
                                    <small>Add todo in each line (press enter for new line)</small>
                                    <textarea
                                    name="desc" 
                                    type="desc"
                                    data-testid="desc"
                                    className="form-control"  
                                    rows="3"
                                    value={desc}
                                    required
                                    onChange={this.onChange}
                                    placeholder="1. todo1"
                                    ></textarea>
                                </div>
                                <button type="submit" data-testid="button" className="btn btn-success">Add</button>
                        </form>
                    </div>
                    <Right/>
                </div>
            </div>
        )
    }
}

Add.propTypes = {
    addTodo: PropTypes.func.isRequired
  };



  const Right = props => {
    return (
      <div
        className="right">
      </div>
    );
  };

export default connect(null, { addTodo})(Add);
