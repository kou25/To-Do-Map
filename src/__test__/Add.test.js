import React from 'react'
import {createStore} from 'redux'
import {Provider} from 'react-redux'
import{render,  cleanup} from '@testing-library/react'
import "@testing-library/jest-dom/extend-expect"
import Add from '../containers/add/Add'
import {ADD_TODO} from '../actions/types'
import renderer from 'react-test-renderer'


afterEach(cleanup)

const startingState={
    name:'',
    desc:''
};
function reducer(state=startingState, action){
  switch(action.type){
    case ADD_TODO:
            return{
                ...state,
                name: state.name.concat(action.payload),
                desc: state.desc.concat(action.payload)
            }
    default:
      return state
  }
}

function renderWithRedux(component, 
  {initialState, store=createStore(reducer, 
    initialState)}={}){
      return{
      ...render(<Provider store={store}>{component}</Provider>)
      }
}

it("renders heading correctly", ()=>{
  const{getByTestId}=render(
    <Provider store={createStore(reducer)}>
    <Add/>
  </Provider>)
  expect(getByTestId('heading')).toHaveTextContent('Add To-DO')
})

it("renders with redux",()=>{
  const{getByTestId}=renderWithRedux(<Add/>)
  expect(getByTestId('name')).toHaveTextContent('')
})

it("renders with redux",()=>{
  const{getByTestId}=renderWithRedux(<Add/>)
  expect(getByTestId('desc')).toHaveTextContent('')
})

it("renders with redux",()=>{
  const{getByTestId}=renderWithRedux(<Add/>)
  expect(getByTestId('button')).toHaveTextContent('Add')
})


it('matches snapshot', ()=>{
  const tree=renderer.create(
        <Provider store={createStore(reducer)}>
          <Add/>
        </Provider>).toJSON()
  expect(tree).toMatchSnapshot()
})









