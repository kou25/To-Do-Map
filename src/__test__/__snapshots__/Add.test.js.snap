// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="container"
>
  <div
    className="d-sm-flex align-items-center justify-content-between mb-4"
  >
    <h1
      data-testid="heading"
      style={
        Object {
          "color": "#8B4513",
          "fontFamily": "cursive",
          "margin": "50px 30%",
          "textShadow": "1px 1px 2px black",
        }
      }
    >
      Add To-DO
      <i
        aria-hidden="true"
        className="fa fa-plus-circle"
        style={
          Object {
            "color": "#8B4513",
            "marginLeft": "20px",
          }
        }
      />
    </h1>
  </div>
  <div
    className="card shadow mb-2"
    style={
      Object {
        "borderRadius": "20px",
        "fontFamily": "cursive",
        "marginLeft": "22%",
        "width": "40%",
      }
    }
  >
    <div
      className="card-body"
    >
      <form
        onSubmit={[Function]}
      >
        <div
          className="form-group"
        >
          <label
            htmlFor="name"
          >
            Pads
          </label>
          <input
            className="form-control"
            data-testid="name"
            name="name"
            onChange={[Function]}
            placeholder="Enter name for to-do"
            required={true}
            type="text"
            value=""
          />
        </div>
        <div
          className="form-group"
        >
          <label
            htmlFor="description"
          >
            To Dos
          </label>
          <br />
          <small>
            Add todo in each line
          </small>
          <textarea
            className="form-control"
            data-testid="desc"
            name="desc"
            onChange={[Function]}
            placeholder="1. todo1"
            required={true}
            rows="3"
            type="desc"
            value=""
          />
        </div>
        <button
          className="btn btn-success"
          data-testid="button"
          type="submit"
        >
          Add
        </button>
      </form>
    </div>
  </div>
</div>
`;
